src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -89,52 +84,28 @@ public virtual string InstanceId
 
         public virtual string ApplicationInsightsInstrumentationKey
         {
-            get => GetSettingFromCache(EnvironmentSettingNames.AppInsightsInstrumentationKey);
-            set => UpdateSettingInCache(EnvironmentSettingNames.AppInsightsInstrumentationKey, value);
+            get => Utility.GetSettingFromConfigOrEnvironment(EnvironmentSettingNames.AppInsightsInstrumentationKey);
```

Comment:
The odd fact that this code would read in a different way than it would store is the reason for the test changes that I had to make. In app.config [here](https://github.com/Azure/azure-functions-host/blob/b4313d45ebf47d9a94cc8a6243204dbeb6240d42/test/WebJobs.Script.Tests/app.config#L160) we're setting this to empty string, and the tests were relying on this cache behavior to be able to update the value.